frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -133,6 +133,43 @@ export default class FirebaseInteractor {
     });
   }
 
+  /**
+   * Creates a new assessment with all of the words from a given intervention.
+   */
+  async createAssessmentFromIntervention(setId: string) {
+    let intervention = await this.db
+      .collection("interventions")
+      .doc(setId)
+      .get();
+    let interventionData = intervention.data();
+    if (interventionData == null) {
+      throw new Error(`Intervention with id ${setId} does not exist`);
+    }
+
+    let wordList = interventionData.wordList;
+    let newAssessment = this.db.collection("assessments").doc();
+
+    for (let word of wordList) {
+      let wordData = await this.getWord(word);
+
+      let newWordData = {
+        correctImage: wordData.correctImage,
+        dateCreated: wordData.createdAt,
+        incorrectImages: wordData.incorrectImages,
+        value: wordData.value,
+      };
+
+      await newAssessment.collection("words").doc(word).set(newWordData);
```

Comment:
I don't think you need to override the word. The word already has these values, so I think you are just overriding this with the same data.

Reply:
Oh right, awkward 😅